threedi_modelchecker/checks/other.py

```diff
@@ -504,3 +504,38 @@ def get_position(point, linestring):
 
     def description(self) -> str:
         return f"{self.column_name} must be more than {self.min_distance} m apart (or exactly on the same position)"
+
+
+class PumpStorageTimestepCheck(BaseCheck):
+    """Check that a pumpstation will not empty its storage area within one timestep"""
+
+    def get_invalid(self, session: Session) -> List[NamedTuple]:
+        return (
+            session.query(models.Pumpstation)
+            .join(
+                models.ConnectionNode,
+                models.Pumpstation.connection_node_start_id == models.ConnectionNode.id,
+            )
+            .filter(
+                (models.ConnectionNode.storage_area != None)
+                & (
+                    (  # calculate how many seconds the pumpstation takes to empty its storage: (storage * height)/pump capacity
+                        (
+                            models.ConnectionNode.storage_area
+                            * (
+                                models.Pumpstation.start_level
+                                - models.Pumpstation.lower_stop_level
+                            )
+                        )
+                        / models.Pumpstation.capacity
+                    )
+                    < Query(models.GlobalSetting.sim_time_step)
```

Comment:
This should compare against a scalar subquery of the *first* global setting, not the global setting with id=1. See how it is done with `kmax` in config.py.